src/main/java/pl/wit/projekt/JPGScanner.java

```diff
@@ -1,40 +1,83 @@
 package pl.wit.projekt;
 
+import java.util.ArrayList;
 import java.util.Date;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.nio.file.attribute.FileTime;
 
 public class JPGScanner {
+	
 	// Data utworzenia -> Lista nazw plików
-	private Map<Date, List<String>> mapMetaFilenames;
+	private Map<FileTime, ArrayList<Path>> filesMetadata;
 
-	public JPGScanner(String sourcePath) {
+	public JPGScanner(Path path) throws IOException {
 		// inicjalizacja mapy
-		mapMetaFilenames = null;
+		filesMetadata = null;
 
-		List<String> listFilenames = scanDirectory(sourcePath);
-		getMetadata(listFilenames);
+		List<Path> sourceFiles = scanDirectory(path);
+		getMetadata(sourceFiles);
 	}
 
 	/**
 	 * Metoda skanująca katalog w poszukiwaniu plików JPG
 	 * 
 	 * @param sourcePath
```

Comment:
Jak kończysz pisać kod to warto zaktualizować komentarze, bo potem musimy z tego wygenerować dokumentację (tu brakuje np. opisu co oznacza zwracana wartość)